gui/gui.py

```diff
@@ -0,0 +1,157 @@
+
+import os
+import tkinter
+import screeninfo
+import time
+import traceback
+if os.name == "nt":
+    import pythoncom
+    import win32com.client
+    import win32gui
+
+from utils.config import USE_GUI, TIMEOUT_GUI
+
+
+def set_foreground_window(wid):
+    if os.name == "nt":
+        win32gui.ShowWindow(wid, 5)
+        try: # this might fail if im switching windows a lot, should be safe to ignore it (instead of crashing the program)
+            win32gui.SetForegroundWindow(wid)
+        except Exception as e:
+            pass
+def get_foreground_window():
+    if os.name == "nt":
+        return win32gui.GetForegroundWindow()
+    return 0
+
+components = []
+
+def close_all_windows():
+    for x in components:
+        x.close()
+
+def check_timeout_gui():
+    for x in components:
+        x.should_close()
+
+def destroy_gui():
+    for x in components:
+        x.destory()
+
+class DisplayWindow:
+    def __init__(self):
+        self.frame = None
+        self.opened = time.time() # When the window was created
+        self.elapsed = 0 # Used to see how long the window was open
+        self.created = False
+        self.prev = None # The Active foreground window before this one
+        self.window_id = 0
+        components.append(self)
+    
+    def prepare_window(self):
+        frame = tkinter.Toplevel()
+        frame.wm_attributes("-topmost", 1)
+        frame.overrideredirect(True)
+        frame.option_add("*Font", "courier 12")
+        frame.withdraw()
+        self.frame = frame
+
+    def destory(self, event = None):
+        self.frame.withdraw()
+        self.frame.destroy()
+        self.frame.update()
+
+    def close(self, event = None):
+        if self.created:
+            self.frame.update()
+            if self.window_id == get_foreground_window():
+                set_foreground_window(self.prev)
+            self.frame.withdraw()
+            for child in self.frame.winfo_children():
+                child.destroy()
+            self.frame.withdraw()
```

Comment:
Why are you calling withdraw twice on frame?

Also, shouldnt you destroy the frame as well? As I said before, there isnt really a point in doing this if you are just going to do same thing as we have currently in master by showing and hiding windows

Reply:
Depends on what we want, i am not sure if we want to create new classes for each type of window or have them in one class,

I will fix the typos

Reply:
I mean, its not about having separate class for each type of window, that is easily doable even on master with really small refactor. But if you are just hiding/showing windows like it is on master, then instead of having just 1 window, the way you approached this is you have a lot of invisible windows in background (1 for each component). I personally like create/destroy a lot more when it actually works (and it did before in your PR, so I am a bit confused why you keep changing it to show/hide, is it bugged?), but if we want to stick with show/hide, they you should stick with 1 master window where you just replace contents like it is on master, and just pass it around to different gui classes if you want it in separate classes.

Reply:
Alright I will add it back in